### PR TITLE
Add `Beatmap search` (osu!, osu!(lazer), web)

### PR DESCRIPTION
**Pull request by TicClick**
closes https://github.com/ppy/osu-wiki/issues/8128

`divisor` is skipped since it's [disabled on the website](https://github.com/ppy/osu-web/blob/67c59d69fcdbd07d44984707058bc1193535898f/app/Libraries/Search/BeatmapsetSearch.php#L308), and in osu!lazer I couldn't figure for the life of me what it does. it finds _something_, but I fail to see any correlation between difficulties found and their beat snap divisor, or timing signature

## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)
- `SKIP_WIKILINK_CHECK` due to the translations of `Guides` containing links with different case https://github.com/ppy/osu-wiki/actions/runs/5756757513/job/15606706895?pr=9828

### CODE REVIEW

**Review by Walavouchey — COMMENTED** [review]
this should be just "Beatmap search" for the article title imo

File: wiki/Beatmap_search/en.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+| `hp`, `dr` | [HP drain rate](/wiki/Beatmap/HP_drain_rate) |
+| `star`, `stars` | [Star rating](/wiki/Beatmap/Star_rating) |
+| `bpm` | [Tempo of the song](/wiki/Music_theory/Tempo) |
+| `length` | [Length](/wiki/Beatmap/Play_time) in seconds |
+| `key`, `keys` | Number of keys (osu!mania and converted beatmaps only) |
+| `status` | Beatmap status. Value can be `ranked`, `approved`, `pending`, `notsubmitted`, `unknown`, or `loved`, or `r`/`a`/`p`/`n`/`u`/`l` for short. |
+| `created` | Beatmap creation date |
+| `ranked` | Beatmap ranking/approval date |
+
+## osu!(lazer)[^lazer-filters]
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:71]:
```suggestion
## Client (lazer)[^lazer-filters]
```

File: wiki/Beatmap_search/en.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+## Example queries
+
+Find beatmaps with low OD, falling within a certain difficulty range:
+
+```
+stars>=7 stars<8 od<8
+```
+
+(osu!) Show ranked Christmas-related beatmaps that have never been attempted:
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:96]:
```suggestion
(Client) Show ranked Christmas-related beatmaps that have never been attempted:
```

File: wiki/Beatmap_search/en.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+unplayed= status=r christmas
+```
+
+(osu!, website) Check if at least one difficulty of [DJ Fresh - Gold Dust](https://osu.ppy.sh/beatmapsets/28107) by ::{ flag=FR }:: [galvenize](https://osu.ppy.sh/users/381444) is downloaded:
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:102]:
```suggestion
(Client, website) Check if at least one difficulty of [DJ Fresh - Gold Dust](https://osu.ppy.sh/beatmapsets/28107) by ::{ flag=FR }:: [galvenize](https://osu.ppy.sh/users/381444) is downloaded:
```

File: wiki/Beatmap_search/en.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+28107
+```
+
+(website) Find beatmaps ranked between August and October 2010 which contain at least a single difficulty made by ::{ flag=GB }:: [Natteke](https://osu.ppy.sh/users/157177):
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:108]:
```suggestion
(Website) Find beatmaps ranked between August and October 2010 which contain at least a single difficulty made by ::{ flag=GB }:: [Natteke](https://osu.ppy.sh/users/157177):
```

File: wiki/Beatmap_search/en.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+ranked>=2010-08 ranked<2010-11 creator=Natteke
+```
+
+## References
+
+[^website-filters]: [`app/Libraries/Search/BeatmapsetQueryParser.php`](https://github.com/ppy/osu-web/blob/67c59d69fcdbd07d44984707058bc1193535898f/app/Libraries/Search/BeatmapsetQueryParser.php)
+[^lazer-filters]: [`osu.Game/Screens/Select/FilterQueryParser.cs`](https://github.com/ppy/osu/blob/270c03235d280ccca3aecea776fb9517635ed695/osu.Game/Screens/Select/FilterQueryParser.cs)
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:117]:
```suggestion
[^website-filters]: [osu!web source code](https://github.com/ppy/osu-web/blob/67c59d69fcdbd07d44984707058bc1193535898f/app/Libraries/Search/BeatmapsetQueryParser.php)
[^lazer-filters]: [osu!(lazer) source code](https://github.com/ppy/osu/blob/270c03235d280ccca3aecea776fb9517635ed695/osu.Game/Screens/Select/FilterQueryParser.cs)
```

File: wiki/Client/Interface/en.md [hunk 6]
```diff
@@ -144,49 +144,13 @@ Sort beatmaps in a certain order.
...
-*Note: You cannot have the chat console or the options sidebar open if you want to search. Otherwise, anything you type will be perceived as chat text or as an options search query.*
-
-Only beatmaps that match the criteria of your search will be shown. By default, any search will be matched against the beatmaps' artists, titles, creators, and tags.
+*Main article: [Beatmap search](/wiki/Beatmap_search)*
 
-In addition to searching these fields, you can use filters to search through other metadata by combining one of the supported filters with a comparison to a value (for example, `ar=9`).
+![](img/search-bar.jpg)
 
-Supported filters:
+The search bar allows for filtering difficulties that match some criteria. By default, osu! runs full-text search, only showing results that contain all of the specified words. In addition to that, it's possible to search by [Approach Rate](/wiki/Beatmap/Approach_rate), [star rating](/wiki/Beatmap/Star_rating), and many more values using a set of filters such as `ar=8 stars>=5`.
```
Comment (Walavouchey) [inline wiki/Client/Interface/en.md:151]:
```suggestion
The search bar allows for filtering difficulties that match some criteria. By default, osu! runs full-text search, only showing results that contain all of the specified words. In addition to that, it is possible to search by [approach rate](/wiki/Beatmap/Approach_rate), [star rating](/wiki/Beatmap/Star_rating), and many more values using a set of filters such as `ar=8` or `stars>=5`.
```

File: wiki/Client/Interface/en.md [hunk 7]
```diff
@@ -144,49 +144,13 @@ Sort beatmaps in a certain order.
...
-| :-: | :-- |
-| `=` or `==` | Equal to |
-| `!=` | Not equal to |
-| `<` | Less than |
-| `>` | Greater than |
-| `<=` | Less than or equal to |
-| `>=` | Greater than or equal to |
-
-You may also enter a difficulty or beatmap ID number in your search to get a single result.
+To search beatmaps, simply start typing when at the song select screen, making sure to close options and the chat console prior to that.
```
Comment (Walavouchey) [inline wiki/Client/Interface/en.md:153]:
```suggestion
To search beatmaps, simply start typing when at the song select screen (without the options or chat panel open.
```

File: wiki/Beatmap_search/en.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+## osu!(lazer)[^lazer-filters]
+
+| Filter | Description |
+| :-: | :-- |
+| `artist` | Name of the artist |
+| `title` | Name of the song |
+| `creator` | Name of the difficulty creator |
+| `ar` | [Approach Rate](/wiki/Beatmap/Approach_rate) |
+| `cs` | [Circle Size](/wiki/Beatmap/Circle_size) |
+| `od` | [Overall Difficulty](/wiki/Beatmap/Overall_difficulty) |
```
Comment (Walavouchey) [inline wiki/Beatmap_search/en.md:80]:
```suggestion
| `ar` | [Approach rate](/wiki/Beatmap/Approach_rate) |
| `cs` | [Circle size](/wiki/Beatmap/Circle_size) |
| `od` | [Overall difficulty](/wiki/Beatmap/Overall_difficulty) |
```

**Review by Walavouchey — APPROVED** [review]

**Comment by cl8n** [comment]
as far as I can tell "divisor" is searching against "BeatDivisor" from .osu ... which is more like an editor setting than anything to do with the map

**Comment by TicClick** [comment]
> as far as I can tell "divisor" is searching against "BeatDivisor" from .osu ... which is more like an editor setting than anything to do with the map

I considered that, but spot-checking a difficulty or two revealed that they had different beat snap divisor in the editor.. perhaps it only takes the highest/lowest set difficulty's setting into account?

**Pull request merged**